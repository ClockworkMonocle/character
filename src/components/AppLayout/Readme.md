Basic layout for having one section static/fixed while another section scrolls underneath.

```js
require('../Type');

// to simulate the document.body or some other container
const wrapperStyle = {
  height: '500px',
};

const navStyle = {
  padding: '8px',
  backgroundColor: 'var(--dark)',
  color: 'var(--white)',
};

const bodyStyle = {
  padding: '8px',
};

<div style={wrapperStyle}>
  <AppLayout
    nav={(
      <div style={navStyle}>
        <Type variant="heading-1" color="white">My Character</Type>
      </div>
    )}
    body={(
      <div style={bodyStyle}>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type variant="display-2">The end</Type>
      </div>
    )}
  />
</div>
```

Can also make it horizontal:

```js
require('../Type');

// to simulate the document.body or some other container
const wrapperStyle = {
  height: '500px',
};

const navStyle = {
  padding: '8px',
  backgroundColor: 'var(--dark)',
  color: 'var(--white)',
  minWidth: '180px',
  height: '100%',
};

const bodyStyle = {
  padding: '8px',
};

<div style={wrapperStyle}>
  <AppLayout
    horizontal
    nav={(
      <div style={navStyle}>
        <Type variant="heading-1" color="white">My Character</Type>
      </div>
    )}
    body={(
      <div style={bodyStyle}>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type>Content!</Type>
        <Type variant="display-2">The end</Type>
      </div>
    )}
  />
</div>
```
