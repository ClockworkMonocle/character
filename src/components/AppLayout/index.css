
.app_layout--container {
  flex-grow: 1;
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: column;
}

.app_layout--body {
  flex-grow: 1;
  overflow-y: auto;
  -webkit-overflow-scrolling: touch;
}

.app_layout--horizontal {
  flex-direction: row;
}

.app_layout--horizontal .app_layout--nav {
  height: 100%;
}
