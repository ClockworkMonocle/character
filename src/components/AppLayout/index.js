import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import './index.css';

export default function AppLayout({ nav, body, horizontal, className, ...props }) {
  const containerCss = cn(
    'app_layout--container',
    horizontal && 'app_layout--horizontal',
    className);
  return (
    <div className={containerCss}>
      <nav className="app_layout--nav">
        {nav}
      </nav>
      <section className="app_layout--body">
        {body}
      </section>
    </div>
  );
}

AppLayout.propTypes = {
  nav: PropTypes.node,
  body: PropTypes.node,
  horizontal: PropTypes.bool,
};

AppLayout.defaultProps = {
  horizontal: false,
};
