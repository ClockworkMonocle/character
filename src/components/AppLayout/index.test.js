import React from 'react';
import { shallow } from 'enzyme';
import AppLayout from './index';

test('<AppLayout /> renders', () => {
  expect(shallow(<AppLayout
    nav={<h1>heyo</h1>}
    body={<p>my body</p>}
  />)).toMatchSnapshot();
});

test('<AppLayout /> renders horizontal', () => {
  expect(shallow(<AppLayout
    horizontal
    nav={<h1>heyo</h1>}
    body={<p>my body</p>}
  />)).toMatchSnapshot();
});
