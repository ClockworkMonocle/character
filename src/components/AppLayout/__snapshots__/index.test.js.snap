// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppLayout /> renders 1`] = `
<div
  className="app_layout--container"
>
  <nav
    className="app_layout--nav"
  >
    <h1>
      heyo
    </h1>
  </nav>
  <section
    className="app_layout--body"
  >
    <p>
      my body
    </p>
  </section>
</div>
`;

exports[`<AppLayout /> renders horizontal 1`] = `
<div
  className="app_layout--container app_layout--horizontal"
>
  <nav
    className="app_layout--nav"
  >
    <h1>
      heyo
    </h1>
  </nav>
  <section
    className="app_layout--body"
  >
    <p>
      my body
    </p>
  </section>
</div>
`;
